fdbserver/masterserver.actor.cpp

```diff
@@ -916,6 +928,13 @@ ACTOR Future<Void> readTransactionSystemState(Reference<MasterData> self,
 	                                                               self->configuration.tLogReplicationFactor);
 	self->tLogGroupRecoveredState = tLogGroupState;
 
+	// Load SS team to TLogGroup assignments. Actually, put into TLogGroup data structures in recruitEverything.
+	RangeResult ssTeamToTLogGroup = wait(self->txnStateStore->readRange(storageTeamIdToTLogGroupRange));
+	for (auto& r : ssTeamToTLogGroup) {
+		auto teamId = decodeStorageTeamIdToTLogGroupKey(r.key);
+		self->storageTeamIdToGroupId[teamId] = BinaryReader::fromStringRef<UID>(r.value, Unversioned());
```

Comment:
This is loading TLog Group data from txnStateStore saved in previous epoch. For the current epoch, the team should probably be assigned to new TLog Groups of this epoch. I am not sure if previous TLog Group information should be preserved, like if recovery still needs the information.

So at least add a TODO here to figure this out.

Reply:
So there are two assignments : TLog -> TLogGroup and SS -> TLogGroup. You mean we should redo the SS Team -> TLogGroup assignment  across recoveries? That is not prefer the previous group it was assigned?

But TLog -> TLogGroup assignment should remain same if possible.

Reply:
Yes, SS team to TLogGroup assignment needs redo across recoveries, because we can't reuse old TLogGroups (since old TLogs have been stopped).

How can you preserve TLog to TLogGroup assignments across recoveries? I thought TLogGroup has its own UID and UIDs of TLogs. So across recoveries, TLogGroups are going to be changed. Additionally, the number of TLogs can change, which affects the number of TLogGroups. So even if TLogGroup's UID can be reused, there is still a problem of mapping group UIDs.

Reply:
Yes, I though I talked about it, but I was trying to fix the issue by trying looking at older tLog UIDs and the new generation in same address and use that. Any new tLogs would be assigned after considering that. But just assuminig new recruitment every recovery simplifies lot of things. I can also remove a bit of code from recovery stuff then. As in we don't need to look at older mapping anymore for now. 

My intuition was that in average case, if old and new TLog happen to be in same server, it would be fewer processes for SS to talk to. I guess premature optimization.

I will update the PR removing all that then.